tls/s2n_client_extensions.c

```diff
@@ -88,6 +91,13 @@ int s2n_client_extensions_send(struct s2n_connection *conn, struct s2n_stuffer *
     if (conn->config->mfl_code != S2N_TLS_MAX_FRAG_LEN_EXT_NONE) {
         total_size += 5;
     }
+    if (conn->config->use_tickets) {
+        if (client_ticket_len != 0 && conn->client_ticket.data == NULL) {
+            S2N_ERROR(S2N_ERR_INVALID_SESSION_TICKET_EXTENSION);
```

Comment:
How this one can happen? Can we validate it when adding a session to client, instead?

Reply:
Below is how we set the session. So, this shouldn't happen. Will remove this check. 

```    
if (session_ticket_len == 0 || session_ticket_len > s2n_stuffer_data_available(from)) {
        S2N_ERROR(S2N_ERR_INVALID_SERIALIZED_SESSION_STATE);
    }

    GUARD(s2n_alloc(&conn->client_ticket, session_ticket_len));
    GUARD(s2n_stuffer_read(from, &conn->client_ticket));
```